packages/grid/data-grid/src/tests/toolbar.DataGrid.test.tsx

```diff
@@ -42,65 +49,125 @@ describe('<DataGrid /> - Toolbar', () => {
     });
-  });
 
-  it('should increase grid density even if toolbar is not enabled', () => {
-    const rowHeight = 30;
-    render(
-      <div style={{ width: 300, height: 300 }}>
-        <DataGrid {...baselineProps} rowHeight={rowHeight} density="compact" />
-      </div>,
-    );
-
-    // @ts-expect-error need to migrate helpers to TypeScript
-    expect(screen.getAllByRole('row')[1]).toHaveInlineStyle({
-      maxHeight: `${Math.floor(rowHeight * COMPACT_DENSITY_FACTOR)}px`,
+    it('should increase grid density even if toolbar is not enabled', () => {
+      const rowHeight = 30;
+      render(
+        <div style={{ width: 300, height: 300 }}>
+          <DataGrid {...baselineProps} rowHeight={rowHeight} density="compact" />
+        </div>,
+      );
+
+      // @ts-expect-error need to migrate helpers to TypeScript
+      expect(screen.getAllByRole('row')[1]).toHaveInlineStyle({
+        maxHeight: `${Math.floor(rowHeight * COMPACT_DENSITY_FACTOR)}px`,
+      });
+    });
+
+    it('should decrease grid density even if toolbar is not enabled', () => {
+      const rowHeight = 30;
+      render(
+        <div style={{ width: 300, height: 300 }}>
+          <DataGrid {...baselineProps} rowHeight={rowHeight} density="comfortable" />
+        </div>,
+      );
+
+      // @ts-expect-error need to migrate helpers to TypeScript
+      expect(screen.getAllByRole('row')[1]).toHaveInlineStyle({
+        maxHeight: `${Math.floor(rowHeight * COMFORTABLE_DENSITY_FACTOR)}px`,
+      });
     });
   });
 
-  it('should decrease grid density even if toolbar is not enabled', () => {
-    const rowHeight = 30;
-    render(
-      <div style={{ width: 300, height: 300 }}>
-        <DataGrid {...baselineProps} rowHeight={rowHeight} density="comfortable" />
-      </div>,
-    );
-
-    // @ts-expect-error need to migrate helpers to TypeScript
-    expect(screen.getAllByRole('row')[1]).toHaveInlineStyle({
-      maxHeight: `${Math.floor(rowHeight * COMFORTABLE_DENSITY_FACTOR)}px`,
+  describe('<DataGrid /> - Column selector', () => {
+    it('should hide "id" column when hiding it from the column selector', () => {
+      const { getByText, getAllByRole } = render(
+        <div style={{ width: 300, height: 300 }}>
+          <DataGrid {...baselineProps} showToolbar />
+        </div>,
+      );
+
+      expect(getAllByRole('columnheader')[0].innerText.trim()).to.equal('id');
+
+      fireEvent.click(getByText('Columns'));
+      fireEvent.click(document.querySelectorAll('[role="tooltip"] [name="id"]')[0]);
+
+      expect(getAllByRole('columnheader')[0].innerText.trim()).to.equal('brand');
+    });
+
+    it('should hide all columns when clicking "HIDE ALL" from the column selector', () => {
+      const { getByText, queryAllByRole } = render(
+        <div style={{ width: 300, height: 300 }}>
+          <DataGrid {...baselineProps} showToolbar />
+        </div>,
+      );
+
+      expect(queryAllByRole('columnheader')).to.not.deep.equal([]);
```

Comment:
here instead of not eq [], I would put what it should be equal to

Reply:
I agree but it's a query selector I didn't want to make it equal to a dom element. I'm opened to suggestions on how to test it. Another way is to check the state and see which column has `hide` and which doesn't.

Reply:
there is this helper function,` getColumnHeaders`

Reply:
Updated